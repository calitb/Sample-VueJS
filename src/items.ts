export interface Item {
  id: string;
  name: string;
}

export function itemImageURL(item?: Item): string {
  if (!item) return '';
  const itemName = item.name
    .replace("'", '')
    .replace('.', '')
    .replace(' ', '-')
    .toLowerCase();
  return `https://img.pokemondb.net/artwork/${itemName}.jpg`;
}
