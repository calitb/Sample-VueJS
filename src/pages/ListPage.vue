<template>
  <div class="flex flex-wrap m-4">
    <article class="p-3 border border-blue-200" v-for="item in $store.state.items" :key="item.id">
      <div class="flex justify-center items-center w-32 h-32">
        <img :src="item | imageSRC" class="w-auto h-24" />
      </div>
      <div class="flex justify-center">
        <router-link :to="{ name: 'detail', params: { detailId: item.id } }">{{ item.name }}</router-link>
      </div>
    </article>
  </div>
</template>

<script lang="ts">
import Vue from "vue";

export default Vue.extend({
  name: "app-lista"
});
</script>
