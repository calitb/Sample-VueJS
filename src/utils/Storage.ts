export function removeItem(key: string) {
  sessionStorage.removeItem(key);
}

export function getItem(key: string): string | null {
  return sessionStorage.getItem(key);
}

export function setItem(key: string, value: string) {
  sessionStorage.setItem(key, value);
}
