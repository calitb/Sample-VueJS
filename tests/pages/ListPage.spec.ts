import { shallowMount, mount, RouterLinkStub } from '@vue/test-utils';
import ListPage from '@/pages/ListPage.vue';
import { itemImageURL, itemsFixture } from '@/items';

import { createLocalVue, createStore } from '../utils';
const localVue = createLocalVue();
localVue.filter('imageSRC', itemImageURL);

describe('List Page', () => {
  describe('Default', () => {
    const store = createStore({ items: itemsFixture });
    const wrapper = shallowMount(ListPage, {
      localVue,
      store,
      stubs: {
        'router-link': true
      }
    });

    it('should render the component', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should render 3 items ', () => {
      const items = wrapper.findAll('b-list-group-item-stub');
      expect(items).toHaveLength(3);
    });

    it('should render the images', () => {
      const items = wrapper.findAll('b-list-group-item-stub');
      const image = items.at(0).find('img');
      expect(image.attributes().src).toBe('https://img.pokemondb.net/artwork/charmander.jpg');
    });

    it('should render the name', () => {
      const items = wrapper.findAll('b-list-group-item-stub');
      const image = items.at(1).find('router-link-stub');
      expect(image.text()).toBe('Charmeleon');
    });
  });

  describe('Links', () => {
    const store = createStore({ items: itemsFixture });
    const wrapper = mount(ListPage, {
      localVue,
      store,
      stubs: {
        RouterLink: RouterLinkStub
      }
    });

    it('should render the link', () => {
      expect(
        wrapper
          .findAllComponents(RouterLinkStub)
          .at(2)
          .props().to
      ).toEqual({ name: 'detail', params: { detailId: '6' } });
    });
  });
});
