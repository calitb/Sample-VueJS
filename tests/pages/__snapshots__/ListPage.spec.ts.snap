// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Page Default should render the component 1`] = `
<div
  class="flex flex-wrap m-4 justify-center"
>
  <router-link-stub
    class="flex flex-col items-center p-3 border border-blue-200 hover:border-blue-800 cursor-pointer"
    tag="article"
    to="[object Object]"
  >
    <div
      class="flex justify-center items-center w-20 h-20 md:w-32 md:h-32"
    >
      <img
        class="w-auto h-16 md:h-24"
        src="charmander_image"
      />
    </div>
     
    <div
      class="flex justify-center text-sm md:text-base w-32 text-center"
    >
      
      Charmander
    
    </div>
  </router-link-stub>
  <router-link-stub
    class="flex flex-col items-center p-3 border border-blue-200 hover:border-blue-800 cursor-pointer"
    tag="article"
    to="[object Object]"
  >
    <div
      class="flex justify-center items-center w-20 h-20 md:w-32 md:h-32"
    >
      <img
        class="w-auto h-16 md:h-24"
        src="charmeleon_image"
      />
    </div>
     
    <div
      class="flex justify-center text-sm md:text-base w-32 text-center"
    >
      
      Charmeleon
    
    </div>
  </router-link-stub>
  <router-link-stub
    class="flex flex-col items-center p-3 border border-blue-200 hover:border-blue-800 cursor-pointer"
    tag="article"
    to="[object Object]"
  >
    <div
      class="flex justify-center items-center w-20 h-20 md:w-32 md:h-32"
    >
      <img
        class="w-auto h-16 md:h-24"
        src="charizard_image"
      />
    </div>
     
    <div
      class="flex justify-center text-sm md:text-base w-32 text-center"
    >
      
      Charizard
    
    </div>
  </router-link-stub>
</div>
`;
